tests/kernel/install_kotd.pm

```diff
@@ -13,20 +13,90 @@ use 5.018;
 use base "opensusebasetest";
 use utils;
 use testapi;
+use File::Basename;
+use MIME::Base64;
 
-# Add kotd repo
-sub add_repos {
-    my $url = get_required_var("INSTALL_KOTD");
-    if ($url =~ /^[\w.]+-[\w.]+$/) {
-        $url = "http://download.suse.de/ibs/Devel:/Kernel:/$url/standard/";
+=head2 grub_version
+
+    grub_version();
+
+If grub2 is installed, return 2; otherwise, return 1.
+
+=cut
+
+sub grub_version {
+    my $ret = script_run('rpm -q grub2');
+    if ($ret == 0) {
+        return 2;
     }
-    zypper_call("--no-gpg-check ar -f '$url' kotd", timeout => 600);
-    zypper_call("--gpg-auto-import-keys ref",       timeout => 1200);
+    return 1;
 }
 
-# Install kotd kernel and reboot
-sub install_from_repo {
-    zypper_call("install --oldpackage --from kotd kernel-default", timeout => 1200);
+
+=head2 download_kernel
+
+  download_kernel($url, $package);
+
+Download package $package from $url with wget and save it to /tmp. C<die> if wget returns non-zero value.
+
+=cut
+
+sub download_kernel {
+    my ($url, $package) = @_;
+    my $kernel = script_output("curl -sk '$url' | grep -oP '$package-[\\d.]+.*?\\.rpm' | head -n1");
+    my $file   = "/tmp/$kernel";
+    if (substr($url, -1) ne '/') {
+        $url = "$url/$kernel";
+    }
+    else {
+        $url = "$url$kernel";
+    }
+    assert_script_run("wget -O '$file' '$url'", timeout => 600);
+    return $file;
+}
+
+=head2 install_kernel
+
+  install_kernel($file);
+
+Install kernel file $file with rpm command and C<die> if it returns non-zero value.
+
+=cut
+
+sub install_kernel {
+    my $file = shift;
+    assert_script_run("rpm -i --nodeps --oldpackage --nosignature '$file'", timeout => 120);
+}
+
+# Set KOTD kernel as default boot option
+=head2 set_default
+
+  set_default($file);
+
+Set $file as the default boot option.
+
+=cut
+
+sub set_default {
+    my $file = shift;
+    my $cmd  = <<'END';
+#!/bin/bash
+rpm -q grub2 &> /dev/null
+if [[ $? -eq 0 ]]; then
+    grub_conf='/boot/grub2/grub.cfg'
+else
+    grub_conf='/boot/grub/menu.lst'
+fi
+
+id=$(echo "$1" | awk -F. '{print $(NF-2)}')
+vmlinuz=$(find /boot -name "vmlinuz-*$id*")
+old_version=$(uname -r)
+new_version=$(basename "$vmlinuz" | sed -e 's/vmlinuz-//g')
+sed -ie "s/$old_version/$new_version/g" "$grub_conf"
+END
+    my $script = "set_default.sh";
+    assert_script_run("cat > '$script' <<'END'\n${cmd}\nEND\ntrue", timeout => 120);
+    assert_script_run("bash '$script' '$file'");
```

Comment:
Don't type scripts into the console. Put them in data/ and download them with curl into the SUT.

And run them with bash -e